ggml/src/ggml-rpc/ggml-rpc.cpp

```diff
@@ -386,6 +386,8 @@ static bool parse_endpoint(const std::string & endpoint, std::string & host, int
 // RPC request : | rpc_cmd (1 byte) | request_size (8 bytes) | request_data (request_size bytes) |
 // No response
 static bool send_rpc_cmd(const std::shared_ptr<socket_t> & sock, enum rpc_cmd cmd, const void * input, size_t input_size) {
+    static std::mutex send_rpc_cmd_mutex;
+    std::lock_guard<std::mutex> lock(send_rpc_cmd_mutex);
```

Comment:
Instead of a global mutex, probably it's better to associate it with the specific instance of the socket. Consider moving the mutex to `struct socket_t`.

Reply:
Good idea, let me try that.